app/src/test/java/cm/aptoide/pt/app/AppViewManagerTest.java

```diff
@@ -0,0 +1,695 @@
+package cm.aptoide.pt.app;
+
+import cm.aptoide.accountmanager.AptoideAccountManager;
+import cm.aptoide.pt.account.view.store.StoreManager;
+import cm.aptoide.pt.analytics.analytics.AnalyticsManager;
+import cm.aptoide.pt.appview.PreferencesManager;
+import cm.aptoide.pt.database.realm.Download;
+import cm.aptoide.pt.database.realm.MinimalAd;
+import cm.aptoide.pt.dataprovider.model.v7.store.Store;
+import cm.aptoide.pt.dataprovider.ws.v2.GenericResponseV2;
+import cm.aptoide.pt.download.AppContext;
+import cm.aptoide.pt.download.DownloadFactory;
+import cm.aptoide.pt.download.InstallType;
+import cm.aptoide.pt.download.Origin;
+import cm.aptoide.pt.install.Install;
+import cm.aptoide.pt.install.InstallAnalytics;
+import cm.aptoide.pt.install.InstallManager;
+import cm.aptoide.pt.notification.NotificationAnalytics;
+import cm.aptoide.pt.search.model.SearchAdResult;
+import cm.aptoide.pt.store.StoreUtilsProxy;
+import cm.aptoide.pt.timeline.SocialRepository;
+import cm.aptoide.pt.view.AppViewConfiguration;
+import cm.aptoide.pt.view.app.AppCenter;
+import cm.aptoide.pt.view.app.AppStats;
+import cm.aptoide.pt.view.app.AppsList;
+import cm.aptoide.pt.view.app.DetailedApp;
+import cm.aptoide.pt.view.app.DetailedAppRequestResult;
+import cm.aptoide.pt.view.app.FlagsVote;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import rx.Completable;
+import rx.Observable;
+import rx.Single;
+
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+/**
+ * Created by D01 on 28/05/2018.
+ */
+
+public class AppViewManagerTest {
+
+  private final int limit = 18;
+  @Mock private InstallManager installManager;
+  @Mock private DownloadFactory downloadFactory;
+  @Mock private AppCenter appCenter;
+  @Mock private ReviewsManager reviewsManager;
+  @Mock private AdsManager adsManager;
```

Comment:
I wouldn't mock this, but i would mock the accessor, the repository inside of it and the mapper (this one is just because of all the request responses types, which would be hard to create).